### PR TITLE
Uyuni users formula with forms

### PR DESCRIPTION
**Pull request by rjmateus**
This formula takes leverage from the new salt states defined on https://github.com/uyuni-project/uyuni/pull/2242 to define a new formula which will be able to create the following object on uyuni servers: 
* Organizations
* System Groups in organizations
* Users in organizations
    * User Groups
    * Access to System Groups
    * Access to software channels

This PR fixes: https://github.com/SUSE/spacewalk/issues/11986

### CODE REVIEW

File: uyuni-users-formula/uyuni-users/init.sls [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{{org['org_id']}}_{{user['name']}}_channels:
+  uyuni.user_channels:
+    - name: {{user['name']}}
+    - password: {{user['password']}}
+    - org_admin_user: {{org['org_admin_user']}}
+    - org_admin_password: {{org['org_admin_password']}}
+    - manageable_channels: {{user.get('manageable_channels',[])}}
+    - subscribable_channels: {{user.get('subscribable_channels', [])}}
+
+    {% endfor %}
```
Comment (renner) [inline uyuni-users-formula/uyuni-users/init.sls:46]: The indentation of those `for` loops reads a bit awkward here as the states are not indented at all. I am currently not sure though what is the best practice regarding that.

File: uyuni-config-formula/uyuni-config/init.sls [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+{% for user in org.get('users', []) %}
+{{org['org_id']}}_{{user['name']}}:
+  uyuni.user_present:
+    - name: {{user['name']}}
+    - password: {{user['password']}}
+    - email: {{user['email']}}
+    - first_name: {{user['first_name']}}
+    - last_name: {{user['last_name']}}
+    - org_admin_user: {{org['org_admin_user']}}
```
Comment (moio) [inline uyuni-config-formula/uyuni-config/init.sls:32]: Nitpick: I'd move `org_admin_user` and `org_admin_password` last in all blocks here, just for uniformity

File: uyuni-config-formula/metadata/metadata.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+description:
+  Create organizations, users, system groups, etc using salt states
```
Comment (moio) [inline uyuni-config-formula/metadata/metadata.yml:2]:
```suggestion
  Create organizations, users, system groups, etc. on an Uyuni or SUSE Manager Server
```

File: uyuni-config-formula/uyuni-config-formula.spec [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+Name:           uyuni-config-formula
+Version:        0.1
+Release:        0
+Summary:        Configure uyuni with organizations, users, groups, etc
+License:        Apache-2.0
+Group:          System/Packages
+Url:            https://github.com/SUSE/salt-formulas
+Source:         uyuni-config-formula-%{version}.tar.gz
+BuildRoot:      %{_tmppath}/%{name}-%{version}-build
+BuildArch:      noarch
```
[the remark this answers is not in the document]
Comment (rjmateus) [reply uyuni-config-formula/uyuni-config-formula.spec:30]: added in last commit

**Comment by rjmateus** [comment]
Thanks for the review. The last commit addresses the comments

**Review by moio — APPROVED** [review]

**Pull request merged**